site/content/blog/2019-07-18-chain-react-2019.md

```diff
@@ -0,0 +1,123 @@
+app the JavaScript engine kicks in, parses the bundle and compiles it into bytecode and finally it 
+gets executed. Hermes changes things a bit, your app is still transpiled with babel and then minified, 
+but Hermes uses an ahead-of-time compiler, which runs as part of the mobile app build process.
+This also allows to spend more time optimizing the bytecode, and with this the bytecode is smaller in 
+size and more efficient. Some of the optimizations that are performed include function deduplication 
+and string table packing.
+
+Basically with Hermes the following is done during build time: transpile with babel, minify, 
+parse and compile to bytecode and then on run time the app just executes, 
+meaning that it starts faster as there are less actions to take when the app first loads.
+
+One thing that I'm not really sure about is what is the impact that using Hermes will cause to CodePush 
+or other OTA (Over the Air) updates. The reality is that we at Mattermost do not use OTA as of now.
+
+Conventional engine vs Hermes
+![Hermes build and run time](/blog/2019-07-18-chain-react-2019/hermes-build.gif)
+
+##### No Just in Time compiler (JIT)
+Most widely used JavaScript engines like JSC use JIT to lazily compile interpreted code to machine code, 
+but Hermes does not have a JIT compiler, meaning that it can underperform some benchmarks, especially 
+those that depend on CPU performance. In reality those benchmarks in general are not representative of 
+mobile application workloads, so we should be good with this. What we need to understand is that 
+conventional engines that use JIT need to warm up when the application starts and that has an impact 
+with TTI (Time to interaction). Also, let's not forget that JIT will also add to code size and memory 
+consumption that will also have a negative impact in cold start. Basically there are some trade-offs 
+by using Hermes instead of JSC or other conventional engines, but the interpreter performance should 
+be good enough.
+
+##### What does this mean during development
+When developing your app, Hermes will not use ahead-of-time compilation, but instead it will generate the 
+bytecode lazily on the device, allowing rapid interaction with Metro. Obviously this means that the 
+lazy-compiled bytecode will not include all the optimizations of a production build. In theory you 
+should expect a more performant app in production than in development.
+
+Another thing to consider is that Hermes focused on ES6 specifications and it should be kept up to 
+date as new specifications emerge. However, one important thing is that because the goal was to keep the 
+size small, some of the language features are not included such as proxies and local `eval()`. 
+[Here is a complete list of unsupported features](https://github.com/facebook/hermes/blob/master/doc/Features.md#excluded-from-support).
+
+We do use `Reflect` in various places throughout our code base at Mattermost, and it is one of the non 
```

Comment:
```suggestion
We do use `Reflect` in various places throughout our code base at Mattermost, and it is one of the
```

Reply:
@amyblais here I'm saying that is a feature that is NOT supported by Hermes

Reply:
nvm